app/gui/src/model/suggestion_database.rs

```diff
@@ -152,14 +265,28 @@ impl SuggestionDatabase {
                 },
                 entry::Update::Remove { id } => {
                     let removed = entries.remove(&id);
-                    if removed.is_none() {
-                        error!(self.logger, "Received Remove event for nonexistent id: {id}");
+                    match removed {
+                        Some(entry) => {
+                            let qualified_name = entry.qualified_name();
+                            qn_to_id_map.warn_if_absent_and_remove(&qualified_name);
+                        }
+                        None => {
+                            let msg = format!(
+                                "Received a suggestion database update event with the 'Remove' \
+                                kind for a nonexistent entry id: {id}."
```

Comment:
What is "kind"? I recognize this term neither in the context of our Rust codebase nor in the context of databases we use. 

EDIT: I think I got it. Did you want to write `"Received a suggestion database 'Remove' event ..."`?